text/0502_aws-vpclattice.md

```diff
@@ -0,0 +1,1026 @@
+   */
+  readonly serviceNetworkId: string;
+}
+```
+
+
+
+### ServiceNetwork
+
+Creates a cloudformation `AWS::VpcLattice::ServiceNetwork`  
+
+A service network is a logical boundary for a collection of services. Services associated with the network can be authorized for discovery, connectivity, accessibility, and observability. To make requests to services in the network, your service or client must be in a VPC that is associated with the service network.
+
+![Service Network](https://docs.aws.amazon.com/images/vpc-lattice/latest/ug/images/service-network.png)
+
+
+The construct will extend ServiceNetworkBase and implement IServiceNetwork.
+
+```typescript
+/**
+ * Create a vpc lattice service network.
+ * Implemented by `ServiceNetwork`.
+ */
+export interface IServiceNetwork extends core.IResource {
+
+  /**
+  * The Amazon Resource Name (ARN) of the service network.
+  */
+  readonly serviceNetworkArn: string;
+  /**
+   * The Id of the Service Network
+   */
+  readonly serviceNetworkId: string;
+  /**
+   * Is this an imported serviceNetwork
+   */
+  readonly imported: boolean;
+  /**
+   * Add Lattice Service
+   */
+  addService(props: AddServiceProps): void;
+  /**
+   * Associate a VPC with the Service Network
+   */
+  associateVPC(props: AssociateVPCProps): void;
+}
+```
+
+The Construct will consume ServiceNetworkProps.
+
+
+```typescript
+export interface ServiceNetworkProps {
+
+  /** The name of the Service Network. If not provided Cloudformation will provide
+   * a name
+   * @default cloudformation generated name
+   */
+  readonly name?: string;
+
+  /**
+   * The type of  authentication to use with the Service Network. The only method avaialble is
+   * AWS_IAM, however in anticipation, it is expected that other types will be added. 
+   * @default 'AWS_IAM'
+   */
+  readonly authType?: AuthType | undefined;
+
+  /**
+   * Logging destinations
+   * @default: no logging
+   */
+  readonly loggingDestinations?: LoggingDestination[];
+
+  /**
+   * Lattice Services that are assocaited with this Service Network
+   * @default no services are associated with the service network
+   */
+  readonly services?: IService[] | undefined;
+
+  /**
+   * Vpcs that are associated with this Service Network
+   * @default no vpcs are associated
+   */
+  readonly vpcs?: ec2.IVpc[] | undefined;
+  /**
+   * An access mode provides 'pre-canned' conditions for the policy
+   * @default no conditions
+   */
+  readonly accessmode?: ServiceNetworkAccessMode | undefined;
+}
+
+```
+`ServiceNetwork` will implement statics in addition to those in IServiceNetwork, to allow the import of the servicenetwork in other stacks ( potentially cross account ). `fromName` provides a lookup to obtain the Id, using a custom resource.  This provides a way to pass a concrete value between cross account stacks.  
+
+```typescript
+public static fromId(scope: constructs.Construct, id: string, props: ImportedServiceNetworkProps ): IServiceNetwork {
```

Comment:
This one was half a cut and paste out of code error  ( we have to get better at this process of creating the API documentation! )..  I'd love to use the generated API.md rather than manually cut and paste it in..

THere are attributes, but if you are importing the service network into a stack, cross account, ( because its been shared to you via RAM ), there is nothing you can do with those attributes. All you can do is associate your vpc, or associate a service to it. and the only attribute we need is the id.